src/Raven.Server/Documents/Sharding/Executors/AllOrchestratorNodesExecutor.cs

```diff
@@ -16,26 +16,26 @@ namespace Raven.Server.Documents.Sharding.Executors;
 public class AllOrchestratorNodesExecutor : AbstractExecutor
 {
     private readonly ServerStore _store;
-    private readonly ShardedDatabaseContext _database;
+    private readonly DatabaseRecord _record;
 
     private ClusterTopology _clusterTopology;
 
     private readonly ConcurrentDictionary<string, RequestExecutor> _current = new ConcurrentDictionary<string, RequestExecutor>();
     private AllNodesExecutorState _state;
 
     // this executor will contact every single node in the cluster
-    public AllOrchestratorNodesExecutor(ServerStore store, ShardedDatabaseContext database) : base(store)
+    public AllOrchestratorNodesExecutor(ServerStore store, DatabaseRecord record) : base(store)
```

Comment:
I think it is better to have here the `ShardedDatabaseContext database`

Reply:
Passing the existing database record (`this`) will pass a context with the old database record, and the shards will not be updated for `AllOrchestratorNodesExecutor` and `ShardExecutor`